resolver/dns/go17.go

```diff
@@ -0,0 +1,35 @@
+// +build go1.6, !go1.8
```

Comment:
the name of file is go17, but the tag of build is go1.6, the name of file should be go16.go?

Reply:
This build tags say go1.6 <= build < go1.8, which essentially means go1.6 and go1.7. So the file name indicates that this build file is for build <= go1.7.